include/graaflib/graph.tpp

```diff
@@ -129,12 +129,26 @@ graph<VERTEX_T, EDGE_T, GRAPH_TYPE_V>::get_neighbors(
 
 template <typename VERTEX_T, typename EDGE_T, graph_type GRAPH_TYPE_V>
 vertex_id_t graph<VERTEX_T, EDGE_T, GRAPH_TYPE_V>::add_vertex(auto&& vertex) {
-  // TODO: check overflow
-  const auto vertex_id{vertex_id_supplier_++};
+  while (has_vertex(vertex_id_supplier_)) {
+    vertex_id_supplier_++;
+  }
+  const auto vertex_id{vertex_id_supplier_};
```

Comment:
[very minor]: Instead of post-increment (`vertex_id_supplier_++`) we could use pre-increment here (`++vertex_id_supplier`). This is slightly faster as post-increment returns the **old** value while pre-increment returns the **new** value. Therefore pre-increment does not need to keep a temporary for the old value.

Since we are not using the return value anyway we can safely do this (arguably very small) optimization :)